src/EnergyPlus/ZoneEquipmentManager.cc

```diff
@@ -5455,6 +5459,164 @@ namespace ZoneEquipmentManager {
+						ZoneSizingInput( ZoneSizIndex ).DOASHighSetpoint = 22.2;
+					}
+					ReportZoneSizingDOASInputs( ZoneSizingInput( ZoneSizIndex ).ZoneName, "NeutralDehumidifiedSupplyAir",
+						ZoneSizingInput( ZoneSizIndex ).DOASLowSetpoint, ZoneSizingInput( ZoneSizIndex ).DOASHighSetpoint );
+				} else if ( ZoneSizingInput( ZoneSizIndex ).DOASControlStrategy == DOACoolSup ) {
+					if ( ZoneSizingInput( ZoneSizIndex ).DOASLowSetpoint == AutoSize &&
+						ZoneSizingInput( ZoneSizIndex ).DOASHighSetpoint == AutoSize ) {
+						ZoneSizingInput( ZoneSizIndex ).DOASLowSetpoint = 12.2;
+						ZoneSizingInput( ZoneSizIndex ).DOASHighSetpoint = 14.4;
+					} else if ( ZoneSizingInput( ZoneSizIndex ).DOASLowSetpoint == AutoSize &&
+						ZoneSizingInput( ZoneSizIndex ).DOASHighSetpoint > 0.0 ) {
+						ZoneSizingInput( ZoneSizIndex ).DOASLowSetpoint = ZoneSizingInput( ZoneSizIndex ).DOASHighSetpoint
+							- 2.2;
+					} else if ( ZoneSizingInput( ZoneSizIndex ).DOASLowSetpoint > 0.0 &&
+						ZoneSizingInput( ZoneSizIndex ).DOASHighSetpoint == AutoSize ) {
+						ZoneSizingInput( ZoneSizIndex ).DOASHighSetpoint = ZoneSizingInput( ZoneSizIndex ).DOASLowSetpoint
+							+ 2.2;
+					}
+					ReportZoneSizingDOASInputs( ZoneSizingInput( ZoneSizIndex ).ZoneName, "ColdSupplyAir",
+						ZoneSizingInput( ZoneSizIndex ).DOASLowSetpoint, ZoneSizingInput( ZoneSizIndex ).DOASHighSetpoint );
+				}
+				if ( ZoneSizingInput( ZoneSizIndex ).DOASLowSetpoint > ZoneSizingInput( ZoneSizIndex ).DOASHighSetpoint ) {
+					ShowSevereError( "For Sizing:Zone = " + ZoneSizingInput( ZoneSizIndex ).ZoneName );
+					ShowContinueError( "... Dedicated Outside Air Low Setpoint for Design must be less than the High Setpoint" );
+					ErrorsFound = true;
+				}
+
+			}
+		}
+		if ( ErrorsFound ) {
+			ShowFatalError( "Errors found in DOAS sizing input. Program terminates." );
+		}
+	}
+
+	void
+		ReportZoneSizingDOASInputs(
+		std::string const & ZoneName, // the name of the zone
+		std::string const & DOASCtrlStrategy, // DOAS control strategy
+		Real64 const DOASLowTemp, // DOAS design low setpoint temperature [C]
+		Real64 const DOASHighTemp // DOAS design high setpoint temperature [C]
+		)
+	{
+
+		// SUBROUTINE INFORMATION:
+		//       AUTHOR         Fred Buhl
+		//       DATE WRITTEN   March 2016
+		//       MODIFIED       na
+		//       RE-ENGINEERED  na
+
+		// PURPOSE OF THIS SUBROUTINE:
+		// This subroutine writes the DOAS Sizing:Zone input for 1 zone to the eio file
+
+		// METHODOLOGY EMPLOYED:
+		// na
+
+		// REFERENCES:
+		// na
+
+		// Using/Aliasing
+		using namespace DataPrecisionGlobals;
+		using DataGlobals::OutputFileInits;
+		using DataStringGlobals::VerString;
+		using General::RoundSigDigits;
+
+		// Locals
+		// SUBROUTINE ARGUMENT DEFINITIONS:
+
+		// SUBROUTINE PARAMETER DEFINITIONS:
+
+		// INTERFACE BLOCK SPECIFICATIONS
+		// na
+
+		// DERIVED TYPE DEFINITIONS
+		// na
+
+		// SUBROUTINE LOCAL VARIABLE DECLARATIONS:
+		static bool MyOneTimeFlag( true );
```

Comment:
Good ideas. Of course this is just a modified copy of ReportSizingOutput.
So similar issues are scattered through the code.

On Tue, Mar 22, 2016 at 8:27 AM, Edwin Lee <notifications@github.com> wrote:

> In src/EnergyPlus/ZoneEquipmentManager.cc
> <https://github.com/NREL/EnergyPlus/pull/5557#discussion_r57006747>:
>
> > +		using DataStringGlobals::VerString;
> > +		using General::RoundSigDigits;
> > +
> > +		// Locals
> > +		// SUBROUTINE ARGUMENT DEFINITIONS:
> > +
> > +		// SUBROUTINE PARAMETER DEFINITIONS:
> > +
> > +		// INTERFACE BLOCK SPECIFICATIONS
> > +		// na
> > +
> > +		// DERIVED TYPE DEFINITIONS
> > +		// na
> > +
> > +		// SUBROUTINE LOCAL VARIABLE DECLARATIONS:
> > +		static bool MyOneTimeFlag( true );
>
> @wfbuhl <https://github.com/wfbuhl> So these static variables inside
> functions cause issues with unit tests. Basically, in unit tests, there is
> no way to go back into the function and reset the value, so the effects can
> fall downstream to the *next* unit test that is run in the same session.
>
> I would:
>
>    - rename this variable to something clear for this application,
>    - put it in the anonymous namespace at the top of this file
>    <https://github.com/NREL/EnergyPlus/blob/94441801dcd7780ba27759c138f50bcc360e1792/src/EnergyPlus/ZoneEquipmentManager.cc#L191>
>    (Fortran-analogy: make it a module variable), and
>    - make sure the value gets reset inside the clear_state function
>    <https://github.com/NREL/EnergyPlus/blob/94441801dcd7780ba27759c138f50bcc360e1792/src/EnergyPlus/ZoneEquipmentManager.cc#L215>
>    already defined for this namespace so the value is reset for any following
>    unit tests that get run.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/NREL/EnergyPlus/pull/5557/files/94441801dcd7780ba27759c138f50bcc360e1792#r57006747>
>